Documents/LearnJs/Constructor(prototype)/js/script.js

```diff
@@ -0,0 +1,94 @@
+// //1
+function Car(make, model, year, color, price) {
+    this.make = make;
+    this.model = model;
+    this.year = year;
+    this.color = color;
+    this.price = price;
+}
+
+function Garage() {
+    this.cars = [];
+    this.addCar = function (make, model, year, color, price) {
+        const newCar = new Car(make, model, year, color, price);
+        this.cars.push(newCar);
+        return "New car was added: " + make + " " + model;
+    }
+
+    this.addCarFromInput = function () {
+        const make = document.getElementById("makeInput").value;
```

Comment:
document.getElementById('makeInput') should be a private property of the class. In your implementation, you repeatedly search the DOM. You need to perform the DOM operation once and store the result as a property inside the class. Then, access the value from this property.

Reply:
const addCarFromInput = function () {
        const make = makeInput.value; 

Now it's a private method and a property of Garage constructor.